core/src/main/java/org/apache/carbondata/core/statusmanager/SegmentUpdateStatusManager.java

```diff
@@ -348,50 +324,67 @@ public boolean isBlockValid(String segName, String blockName) {
           return deleteFileList;
         }
         final long deltaEndTimeStamp = getEndTimeOfDeltaFile(extension, block);
-
-        // final long deltaEndTimeStamp = block.getDeleteDeltaEndTimeAsLong();
-        // final long deltaStartTimestamp = block.getDeleteDeltaStartTimeAsLong();
-        return getFilePaths(blockDir, blockNameFromTuple, extension, deleteFileList,
-            deltaStartTimestamp, deltaEndTimeStamp);
+        // If start and end time is same then it has only one delta file so construct the file
+        // directly with available information with out listing
+        if (block.getDeleteDeltaStartTimestamp().equals(block.getDeleteDeltaEndTimestamp())) {
+          deleteFileList.add(
+              blockDir + CarbonCommonConstants.FILE_SEPARATOR + block.getBlockName() + "-" + block
```

Comment:
suggest to use `String.format` to construct the path without string concatenation

Reply:
Ok, For better performance and memory utilization StringBuilder is better for simple concatenation. So I used StringBuilder.